presto-main/src/main/java/com/facebook/presto/execution/scheduler/filter/NodeFilter.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.execution.scheduler.filter;
+
+import com.facebook.presto.spi.Node;
+
+import java.util.Set;
+
+public interface NodeFilter<T extends Node>
+{
+    /**
+     * Method to filter a set of nodes matching the input criteria
+     *
+     * @param nodes Nodes to filter
+     * @param nodeSelectionCriteria Criteria to use for filtering nodes
+     * @return filtered nodes
+     */
+    Set<T> filter(Set<T> nodes, NodeSelectionCriteria nodeSelectionCriteria);
+}
```

Comment:
Wondering if we need this interface since the filter is going to be applied on a collection of nodes.
Can we just use `java.util.function.Predicate` instead and create implementations of that.

Reply:
I like it, I think we can use java.util.function.Predicate<Node>